packages/expo-dev-launcher/bundle/components/EmptyBranchesMessage.tsx

```diff
@@ -0,0 +1,86 @@
+import { Spacer, View, Text, Button, Heading } from 'expo-dev-client-components';
+import * as React from 'react';
+
+import { useBuildInfo } from '../providers/BuildInfoProvider';
+import { Branch } from '../queries/useBranchesForApp';
+
+type EmptyBranchesMessageProps = {
+  branches: Branch[];
+  incompatibleBranches: Branch[];
+  onLearnMorePress: () => void;
+};
+
+export function EmptyBranchesMessage({
+  branches = [],
+  incompatibleBranches = [],
+  onLearnMorePress,
+}: EmptyBranchesMessageProps) {
+  const { runtimeVersion } = useBuildInfo();
+
+  // no compatible branches
+  if (branches.length === 0 && incompatibleBranches.length > 0) {
+    const recentBranchWithUpdates = incompatibleBranches.find(
+      (branch) => branch.updates.length > 0
+    );
+    const latestRuntimeVersion = recentBranchWithUpdates?.updates?.[0].runtimeVersion;
+
+    return (
+      <View padding="medium" rounded="large" bg="default">
+        <View>
+          <Heading>The runtime version of this app has no compatible branches.</Heading>
+          <Spacer.Vertical size="small" />
+          <Text
+            color="secondary"
+            size="small">{`However there are ${incompatibleBranches.length} branches that will work with a different runtime version - you might need to update your development build.`}</Text>
```

Comment:
```suggestion
            size="small">{`However, there are ${incompatibleBranches.length} branches that are compatible with a different runtime version. You may need to publish a new update or install a new development build.`}</Text>
```
thinking through this, I think this situation is actually more likely to occur if the dev build is _newer_ than any published update (since older builds would still be able to run older updates on existing branches), and in those cases the fix would be to publish a new update rather than installing a new build -- think that might be worth reflecting in the text

Reply:
that makes a lot of sense